### PR TITLE
Simplify workload requirements specification

### PR DESCRIPTION
**Pull request by rbradford**
Simplify the way that the requirements to run a workload are specified in the payload of an SSNTP command. Replace the slice of resource requests with a simplified structure capturing the requirements to run the workload which can be further extended as necessary and later exposed directly to users in the workload definition.

This is enabling work to allow users to select a node to host their workload by node_id and hostname. The implementation of that requires the exposure of the new requirements as part of the types.Workload definition and will follow later.

### CODE REVIEW

File: ciao-launcher/tests/examples/start_legacy_volume.yaml [hunk 1]
```diff
@@ -1,10 +1,8 @@
 ---
 start:
-  requested_resources:
-     - type: vcpus
-       value: 2
-     - type: mem_mb
-       value: 370
+  requirements:
+    vcpus: 2
+    mem_mb: 256
```
Comment (markdryan) [inline ciao-launcher/tests/examples/start_legacy_volume.yaml]: Was there a reason for changing this value?

File: ciao-launcher/tests/examples/start_nn.yaml [hunk 2]
```diff
@@ -3,13 +3,10 @@
...
-  requested_resources:
-     - type: vcpus
-       value: 2
-     - type: mem_mb
-       value: 256
-     - type: network_node
-       value: 1
+  requirements:
+    vcpus: 2
+    mem_mb: 370
```
Comment (markdryan) [inline ciao-launcher/tests/examples/start_nn.yaml]: Was there a reason for changing this value?

File: ciao-scheduler/scheduler.go [hunk 3]
```diff
@@ -471,104 +471,33 @@ func (sched *ssntpSchedulerServer) StatusNotify(uuid string, status ssntp.Status
...
 
-func networkDemandsSatisfied(node *nodeStat, workload *workResources) bool {
-	if !node.isNetNode {
-		return true
-	}
-
-	var matchedNetworksCount int
-	var requestedNetworksCount int
-
-	for _, requestedNetwork := range workload.physNets {
```
Comment (markdryan) [inline ciao-scheduler/scheduler.go]: Was this stuff simply not used?

File: ciao-launcher/tests/examples/start_legacy_volume.yaml (hunk 1 above)
Comment (rbradford) [reply ciao-launcher/tests/examples/start_legacy_volume.yaml]: Just a copy and paste flail.

File: ciao-launcher/tests/examples/start_nn.yaml (hunk 2 above)
Comment (rbradford) [reply ciao-launcher/tests/examples/start_nn.yaml]: Again just a copy and paste error.

File: ciao-scheduler/scheduler.go (hunk 3 above)
Comment (rbradford) [reply ciao-scheduler/scheduler.go]: With the exception of the !node.isNetNode() which is now done in the caller, yep this code was not used.

**Pull request merged**